picard/ui/scripteditor.py

```diff
@@ -0,0 +1,428 @@
+        else:
+            self.override = override
+        override['file_naming_format'] = self.ui.file_naming_format.toPlainText()
+
+        self.ui.example_filename_before.clear()
+        self.ui.example_filename_after.clear()
+        self.current_row = -1
+
+        examples = self.examples.get_examples(override)
+        for before, after in sorted(examples, key=lambda x: x[1]):
+            self.ui.example_filename_before.addItem(before)
+            self.ui.example_filename_after.addItem(after)
+
+        if send_signal:
+            self.signal_update.emit()
+
+    def toggle_wordwrap(self):
+        if self.ui.file_naming_word_wrap.isChecked():
+            self.wordwrap = QtWidgets.QTextEdit.WidgetWidth
+        else:
+            self.wordwrap = QtWidgets.QTextEdit.NoWrap
+        self.ui.file_naming_format.setLineWrapMode(self.wordwrap)
+
+    def import_script(self):
+        options = QtWidgets.QFileDialog.Options()
+        options |= QtWidgets.QFileDialog.DontUseNativeDialog
+        filename, _ = QtWidgets.QFileDialog.getOpenFileName(self, "Import Script File", "", "All Files (*);;Picard Script Files (*.pts)", options=options)
+        if filename:
+            log.debug('Importing naming script file: %s' % filename)
+            file_text = ""
+            try:
+                with open(filename, 'r', encoding='utf8') as i_file:
+                    file_text = i_file.read()
+            except OSError as error:
+                self.display_error(error)
+                return
+            if file_text.strip():
+                self.ui.file_naming_format.setPlainText(file_text.strip())
+            else:
+                self.ui.file_naming_format.setPlainText("$noop( The import file was empty. )")
+
+    def export_script(self):
+        script_text = self.get_script()
+        if script_text:
+            options = QtWidgets.QFileDialog.Options()
+            options |= QtWidgets.QFileDialog.DontUseNativeDialog
+            filename, _ = QtWidgets.QFileDialog.getSaveFileName(self, "Export Script File", "picard_naming_script.pts", "All Files (*);;Picard Script Files (*.pts)", options=options)
+            if filename:
+                log.debug('Exporting naming script file: %s' % filename)
+                try:
+                    with open(filename, 'w', encoding='utf8') as o_file:
+                        o_file.write(self.get_script() + '\n')
+                except OSError as error:
+                    self.display_error(error)
+                    return
+
+    def load(self):
+        config = get_config()
+        self.toggle_wordwrap()
+        self.ui.file_naming_format.setPlainText(config.setting["file_naming_format"])
+        self.update_examples()
+
+    def check(self):
+        self.check_format()
+
+    def check_format(self):
+        config = get_config()
+        parser = ScriptParser()
+        try:
+            parser.eval(self.ui.file_naming_format.toPlainText())
```

Comment:
Use `get_script()`?

```python
script_text = self.get_script()
try:
    parser.eval(script_text)
except Exception as e:
    raise ScriptCheckError("", str(e))
if not script_text and config.setting["rename_files"]:
    raise ScriptCheckError("", _("The file naming format must not be empty."))
```

Reply:
Much better.  `self.get_script()` was also a later addition, and I didn't think to use it internally.